pyxem/utils/plotting_utils.py

```diff
@@ -113,3 +118,187 @@ def plot_template_over_pattern(
         color=marker_color,
     )
     return (ax, im, sp)
+
+
+def generate_template_markers(
+    signal,
```

Comment:
Is there a reason that `signal` and `library` aren't typed?  If this is a circular import situation you can always use.

Reply:
Just to avoid importing just for type hinting, really. It seemed more in line with the rest of pyxem, which is generally not annotated.
There are no circular imports when adding annotations, but I am aware of `typing.TYPE_CHECKING` if that is what you meant?

Reply:
@viljarjf ahh sorry I meant to delete that last part :) 

I think you should be able to import the class without really causing any slow downs,  in most cases we have to import all of the hyperspy classes at some point. 

> It seemed more in line with the rest of pyxem, which is generally not annotated.

I think at this point I've come around to  type annotation, so this is more a matter of time and effort and less a matter of stylistic choice.